File: oeml-sdk/javascript-closure-angular/API/Client/NewOrder.js
goog.provide('API.Client.NewOrder');

/**
 * @record
 */
API.Client.NewOrder = function() {}

/**
 * Exchange identifier.
 * @type {!string}
 * @export
 */
API.Client.NewOrder.prototype.exchangeId;

/**
 * Unique identifier for the order assigned by the `OEML API` client.
 * @type {!string}
 * @export
 */
API.Client.NewOrder.prototype.clientOrderId;

/**
 * Exchange symbol. One of the properties (`symbol_exchange`, `symbol_coinapi`) is required to identify the market for the order.
 * @type {!string}
 * @export
 */
API.Client.NewOrder.prototype.symbolExchange;

/**
 * CoinAPI symbol. One of the properties (`symbol_exchange`, `symbol_coinapi`) is required to identify the market for the order.
 * @type {!string}
 * @export
 */
API.Client.NewOrder.prototype.symbolCoinapi;

/**
 * Order quantity.
 * @type {!number}
 * @export
 */
API.Client.NewOrder.prototype.amountOrder;

/**
 * Order price.
 * @type {!number}
 * @export
 */
API.Client.NewOrder.prototype.price;

/**
 * @type {!API.Client.OrdSide}
 * @export
 */
API.Client.NewOrder.prototype.side;

/**
 * @type {!API.Client.OrdType}
 * @export
 */
API.Client.NewOrder.prototype.orderType;

/**
 * @type {!API.Client.TimeInForce}
 * @export
 */
API.Client.NewOrder.prototype.timeInForce;

/**
 * Expiration time. Conditionaly required for orders with time_in_force = `GOOD_TILL_TIME_EXCHANGE` or `GOOD_TILL_TIME_OEML`.
 * @type {!API.Client.date}
 * @export
 */
API.Client.NewOrder.prototype.expireTime;

/**
 * Order execution instructions are documented in the separate section: <a href=\"#oeml-order-params-exec\">OEML / Starter Guide / Order parameters / Execution instructions</a>
 * @type {!Array<!string>}
 * @export
 */
API.Client.NewOrder.prototype.execInst;

/** @enum {string} */
API.Client.NewOrder.Array&lt;!ExecInstEnum&gt; = { 
  MAKER_OR_CANCEL: 'MAKER_OR_CANCEL',
  AUCTION_ONLY: 'AUCTION_ONLY',
  INDICATION_OF_INTEREST: 'INDICATION_OF_INTEREST',
}

File: oeml-sdk/clojure/src/oeml_rest_api/specs/new_order.clj
(ns oeml-rest-api.specs.new-order
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [oeml-rest-api.specs.ord-side :refer :all]
            [oeml-rest-api.specs.ord-type :refer :all]
            [oeml-rest-api.specs.time-in-force :refer :all]
            )
  (:import (java.io File)))


(def new-order-data
  {
   (ds/req :exchange_id) string?
   (ds/req :client_order_id) string?
   (ds/opt :symbol_exchange) string?
   (ds/opt :symbol_coinapi) string?
   (ds/req :amount_order) float?
   (ds/req :price) float?
   (ds/req :side) ord-side-spec
   (ds/req :order_type) ord-type-spec
   (ds/req :time_in_force) time-in-force-spec
   (ds/opt :expire_time) inst?
   (ds/opt :exec_inst) (s/coll-of string?)
   })

(def new-order-spec
  (ds/spec
    {:name ::new-order
     :spec new-order-data}))
